src/lib/client.ts

```diff
@@ -106,16 +106,19 @@ function clearStored() {
 
 //TODO should be refactored to not rely on the .init() method to be valid
 export class Client {
-  protected wsProvider = new WsProvider
-  protected api: ApiPromise = new ApiPromise
-  protected farmed = getStoredBlocks()
-  protected clearTauriDestroy: event.UnlistenFn = () => { }
-  protected unsubscribe: event.UnlistenFn = () => { }
-  data = reactive(emptyData)
+  protected wsProvider = new WsProvider("ws://localhost:9944");
+  protected api: ApiPromise = new ApiPromise({ provider: this.wsProvider });
+  protected farmed = getStoredBlocks();
+  protected clearTauriDestroy: event.UnlistenFn = () => { };
+  protected unsubscribe: event.UnlistenFn = () => { };
+  data = reactive(emptyData);
+
   status = {
     farming: () => { }, // TODO return some farming status info
     plot: () => { }, // TODO return some plot status info
-    net: () => { } // TODO return some net status info
+    net: () => {
+      return this.api.rpc.system.peers()
+    } // TODO return some net status info
   }
```

Comment:
Can we have some type information here? I don't see anything and wonder how it even compiles :thinking:

Reply:
The return type is inferred by ts similar to assigning a variable without explicitly defining the variable type. I looked for an option for force explicit return type that could be placed in tsconfig but didn't see it. I do see an option to type check function returns in eslint so I'll try and enable that.
![image](https://user-images.githubusercontent.com/2530957/135732086-c04c7cae-bf5f-4e62-a916-efcb5223d7e1.png)
https://github.com/bradzacher/eslint-plugin-typescript/blob/master/docs/rules/explicit-function-return-type.md

Reply:
There is a big difference between variables and functions/methods. For variables it is fine to infer types, but for functions input and output types should be defined explicitly (the exception is anonymous callbacks whose type is defined in the argument already).

Reply:
I got the eslint change implemented and I'm pushing the changes on another branch right now.

Reply:
Ok I implemented improved checking in another branch that will force functions to have an explicit return type defined.
#20